analyzers/tests/SonarAnalyzer.UnitTest/Common/RuleTest.cs

```diff
@@ -122,129 +107,147 @@ public void SonarDiagnosticAnalyzer_IsUsedInAllRules()
-                .ForEach(diagnostic => diagnostic.CustomTags.Should().Contain(LanguageNames.VisualBasic));
-        }
+        public void AllVbNetRules_HaveVbNetTag() =>
+            SupportedDiagnostics(AnalyzerLanguage.VisualBasic).Should().OnlyContain(diagnostic => diagnostic.CustomTags.Contains(LanguageNames.VisualBasic));
 
         [TestMethod]
         public void AllRules_SonarWayTagPresenceMatchesIsEnabledByDefault()
         {
-            var allAnalyzers =new RuleFinder().AllAnalyzerTypes.SelectMany(type => GetSupportedDiagnostics(type));
-
-            var parameterizedAnalyzers = new RuleFinder().AllAnalyzerTypes
+            var parameterized = new RuleFinder().AllAnalyzerTypes
                 .Where(RuleFinder.IsParameterized)
-                .Select(type => (DiagnosticAnalyzer)Activator.CreateInstance(type))
-                .SelectMany(analyzer => analyzer.SupportedDiagnostics)
+                .SelectMany(type => ((DiagnosticAnalyzer)Activator.CreateInstance(type)).SupportedDiagnostics)
                 .ToHashSet();
 
-            foreach (var analyzer in allAnalyzers)
+            foreach (var diagnostic in new RuleFinder().AllAnalyzerTypes.SelectMany(SupportedDiagnostics))
             {
-                var isInSonarWay = analyzer.CustomTags.Contains(DiagnosticDescriptorBuilder.SonarWayTag);
-
-                if (IsSecurityHotspot(analyzer))
+                if (IsSecurityHotspot(diagnostic))
                 {
                     // Security hotspots are enabled by default, but they will report issues only
                     // when their ID is contained in SonarLint.xml
-                    analyzer.IsEnabledByDefault.Should().BeTrue($"{analyzer.Id} should be enabled by default");
+                    diagnostic.IsEnabledByDefault.Should().BeTrue($"{diagnostic.Id} should be enabled by default");
+                }
+                else if (IsDeprecated(diagnostic))
+                {
+                    // Deprecated rules should be removed from SonarWay
+                    diagnostic.IsEnabledByDefault.Should().BeFalse($"{diagnostic.Id} is deprecated and should be disabled by default (removed from SonarWay)");
                 }
-                else if (parameterizedAnalyzers.Contains(analyzer))
+                else if (parameterized.Contains(diagnostic))
                 {
                     // Even if a a parametrized rule is in Sonar way profile, it is still disabled by default.
                     // See https://github.com/SonarSource/sonar-dotnet/issues/1274
-                    analyzer.IsEnabledByDefault.Should().BeFalse($"{analyzer.Id} has parameters and should be disabled by default");
+                    diagnostic.IsEnabledByDefault.Should().BeFalse($"{diagnostic.Id} has parameters and should be disabled by default");
                 }
-                else if (isInSonarWay)
+                else if (diagnostic.CustomTags.Contains(DiagnosticDescriptorBuilder.SonarWayTag))
```

Comment:
Shouldn't this check run for all rules?

Reply:
Rules that are not in SonarWay are not activated by default in VS. When using SL-only or NuGet, user needs to explicitly activate them.

Reply:
🤦 only now I've seen the comment: `Even if a a parametrized rule is in Sonar way profile, it is still disabled by default.`